node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -107,46 +62,243 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 		})
 		.collect::<Vec<_>>();
 
-	let generics = quote! {
-		< S, #( #baggage_generic_ty, )* >
-	};
-	let where_clause = quote! {
+	// Helpers to use within quote! macros
+	let subsystem_where_clause = &info
+		.subsystems()
+		.iter()
+		.map(|subsys| {
+			let generic_ty = &subsys.generic;
+			let consumes = &subsys.consumes;
+			quote! {
+				#generic_ty : #support_crate ::Subsystem<#subsystem_ctx_name< #consumes >, #error_ty>
+			}
+		})
+		.collect::<Vec<_>>();
+	let builder_where_clause = quote! {
 		where
-			S: #support_crate ::SpawnNamed,
+			#( #subsystem_where_clause, )*
+			S: #support_crate ::SpawnNamed + Send
```

Comment:
nit: is there a reason we require send here?

Reply:
It is also required in the setter: https://github.com/paritytech/polkadot/blob/master/node/overseer/overseer-gen/proc-macro/src/impl_builder.rs#L252

Please correct me if I'm wrong but it seems that if `Send` is required in a setter signature it should also be required in the common `where` clause. Or `Send` could be removed from the both places...

Reply:
I don't quite remember why this was added. We do spawn the subsystems on a threadpool, but at that point the generics are already specialized/known :monocle_face: I'd say get the tests to pass and then let's try to remove it.

Reply:
Leave as is for now.